src/node_state.rs

```diff
@@ -269,167 +304,68 @@ impl NodeState {
     }
 
     fn verify_hyle_output(
-        &self,
+        unsettled_tx: &mut UnsettledBlobTransaction,
         contract_name: &ContractName,
         hyle_output: &HyleOutput,
         unsettled_tx_hash: &TxHash,
     ) -> Result<(), Error> {
+        // TODO: this is perfectly fine and can be settled, and should be removed.
         if !hyle_output.success {
             bail!("Contract execution is not a success");
         }
```

Comment:
then why don't you remove it ?

Reply:
Je comptais le faire dans une PR ultérieure, mais ça peut se combiner. Sais pas encore exactement les changements qui seraient nécessaires.